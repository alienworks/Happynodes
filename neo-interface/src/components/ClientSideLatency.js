import axios from 'axios'
import React, { Component } from 'react'
import Ping from 'ping.js';

class ClientSideLatency extends Component {
    constructor(props) {
        super(props);
        this.state = {
            latency: 0
        }

        this.getLatency = this.getLatency.bind(this);
    }

    getLatency() {
        var p = new Ping();
        const url = this.props.url;

        var t0 = performance.now();

        p.ping(url, (err, data) => {
            var t1 = performance.now();
            this.setState({ latency: (t1 - t0) });
        });
    }

    componentDidMount() {
        this.getLatency()
    }

    render() {
        return (
            <div className="infoblock">
                <h4>{(this.state.latency.toFixed(0)).toLocaleString()}</h4>
                <h5>Live Client Side Latency (ms)</h5>
            </div>
        );
    }
}

export default ClientSideLatency;