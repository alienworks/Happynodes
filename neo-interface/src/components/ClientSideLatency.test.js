import React from 'react';
import ReactDOM from 'react-dom';
import ClientSideLatency from './ClientSideLatency';
import renderer from 'react-test-renderer';

Date.now = jest.fn(() => 1487076708000)

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ClientSideLatency />, div);
    ReactDOM.unmountComponentAtNode(div);
});

describe("test against valid snapshot", ()=> {
    it("redner as expected", ()=> {
        const tree = renderer.create(<ClientSideLatency url="http://api.otcgo.cn:10332" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
  