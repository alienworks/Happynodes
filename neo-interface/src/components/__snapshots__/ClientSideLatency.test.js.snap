// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test against valid snapshot redner as expected 1`] = `
<div
  className="infoblock"
>
  <h4>
    0
  </h4>
  <h5>
    Live Client Side Latency (ms)
  </h5>
</div>
`;
